aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/BaseHandlerStd.java

```diff
@@ -87,10 +87,13 @@ protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final ProxyClient<RdsClient> proxyClient,
             final Logger logger);
 
-    protected ProgressEvent<ResourceModel, CallbackContext> updateOptionGroup(
+    protected ProgressEvent<ResourceModel, CallbackContext> updateOptionGroupIfRequired(
```

Comment:
I personally am not a big fan of `callIfCondition` methods. Another alternative to that would be:
`if condition { call(); }`. The latter option is semantically cleaner. The condition could be simplified with a supplementary method, like: `isShouldCall()`.